### ISSUE TITLE
Admin interface: Schedule: don't auto-save

### ISSUE THREAD

**Issue by crista** [clowdr-app/clowdr#47]
Auto-save here us a nuisance! Consider this use case:

- Create a new event. This creates an entry with times that default to the current time. The entry ends up somewhere on the table.
- Change the day/time. Poof! the entry suddenly disappears out of focus and goes somewhere else on the table. We need to visually search for it.

Here's what I suggest:

- Creating a new event places a new entry at the very top of the table
- Let me edit it to my heart's content, and then let me press a Save button
- (Ideally, but not as important as the last two things) If the entry changed places on the table, show me where it went

**Comment by EdNutting** [comment]
* We know focus control needs a bit of work
* Creation will be handled via a modal in future. Having it instantly create a row is not ideal for a number of reasons (inc. the one you've mentioned)

* Placing an entry at the top of the table is basically impossible. Our choice was "use an open source react component to build the tables" or "attempt a third time to write our own". We picked the former because it has a lot more features (like proper sorting and grouping). Unfortunately, placing a row at the start is not one such feature.
* Instant save is much simpler for most users. We have had too many organisers forget to click save. Batch saving also poses significant problems for consistency in the presence of multiple editors (as we will have to handle at larger conferences like ICSE). We will be sticking with instant saving.
* Again, focus control needs some work.

**Comment by EdNutting** [comment]
* Focus control is now resolved.
* Creation is now handled via a modal
* Creation inserts the row into the bottom of the current page without refreshing everything else. Focus is maintained on the create button when the modal closes.
* Instant-saving no longer messes up focus control

(I wrote an entirely new CRUD Table component to tackle many issues including these)

I believe the only remaining missing feature is "undo".
